client/lib/cart/store/cart-analytics.js

```diff
@@ -33,3 +33,7 @@ function recordAddEvent( cartItem ) {
 function recordRemoveEvent( cartItem ) {
 	analytics.tracks.recordEvent( 'calypso_cart_product_remove', removeNestedProperties( cartItem ) );
 }
+
+export function recordUnrecognizedPaymentMethod( action ) {
+	analytics.tracks.recordEvent( 'calypso_cart_unrecognized_payment_method', action );
```

Comment:
Hmm, from your screenshot, it looks like it indeed had worked on your end at some points. However, I don't think it has ever worked for nested properties, so I was not surprised when I saw this:
![image](https://user-images.githubusercontent.com/1842898/49640963-d81aed00-fa49-11e8-9edb-cc77d88bdda4.png)
I broke the switch so that `recordUnrecognizedPaymentMethod()` would always be called.

Working with a nested object or not, I'd suggest we only record necessary props for this event so that we don't submit too much noise. At very least we should omit the action type.

Reply:
After 0f72c00fdec775d68c1190791da261b51fc7588b, the object is stringified, so it works. 

<img width="1748" alt="screen shot 2018-12-10 at 10 42 34 am" src="https://user-images.githubusercontent.com/6458278/49704553-7be1e400-fc68-11e8-8d29-de09c08cb2fa.png">

Assuming the entire action object is useful for tracking?

Reply:
Hmm, to think it again, since Tracks events are more for behavior analysis, I think we should consider to use [log2logstash](https://github.com/Automattic/wp-calypso/blob/master/client/state/logstash/actions.js) instead in this case. See "When should I use a different tracking/logging tool" section in p2-PCYsg-4D1. Using logstash also makes it easier to search.

Does that make sense to you?

Reply:
> should consider to use log2logstash instead in this case

**TIL**

Reply:
I did actually look at logstash first, and noticed that calypso comment about not using it, and then finally realized it wasn't working because there's a 1 element white-list on the back end. I wasn't confident enough in my assessment to go that way, so I switched to tracks, which would work without backend changes.

Since then, though, I found out that we can easily set up alerts on tracks, so now I'd prefer to go that way (unless we can get logstash alerts too, and I just don't know about it?)

Reply:
> Assuming the entire action object is useful for tracking?

Yeah, we need to add code to extract the billing address for every payment method, current and future, so passing the payment tells gives us the shape of the data so we can handle it. Hopefully we'll never see an action without a `payment` field, but if we ever did the combination of weird and rare makes me want all the info we can get.

Reply:
Sounds reasonable to me.